kyo-prelude/shared/src/main/scala/kyo/Guard.scala

```diff
@@ -0,0 +1,68 @@
+package kyo
+
+import kyo.Guard.CanGuard
+import kyo.kernel.ArrowEffect
+
+/** Protect effect(s) [[S]] from being handled. To be used in functions that accept effects with generic effect types in order to avoid
+  * accidentally handling them. Can only be used within [[Guard.use]] blocks.
+  *
+  * @tparam S
+  *   The effect type that is guarded
+  * @see
+  *   [[Guard]]
+  */
+sealed trait Guarded[S] extends ArrowEffect[[A] =>> A < S, Id]
```

Comment:
this is quite useful! would it cover the `Abort` case as well? I think not because of the erased tags, right?

Reply:
Maybe `Mask` for the name could work better? It seems to express well what the effect does since it masks the effect tags. We have `fiber.mask` but it's a rare low-level operation so the name isn't use yet